datasets/li/posting_sanctions/crawler.py

```diff
@@ -77,88 +75,98 @@ def parse_target(
         context.emit(person, target=True)
         return person
 
-    company = context.make("Company")
-    company.id = context.make_id("Company", company_name)
-    company.add("name", company_name)
-    h.apply_address(context, company, address)
-
     emp = context.make("Employment")
     emp.id = context.make_id("Employment", person.id, company.id)
     emp.add("employee", person)
     emp.add("employer", company)
-    h.apply_date(emp, "date", date)
+    emp.add("date", date)
     emp.add("role", "Manager found responsible for breaking the law")
     context.emit(person, target=False)
     context.emit(emp, target=False)
 
     return company
 
 
-def parse_debarments(context: Context, doc) -> None:
-    table = doc.xpath(
-        "//h2[text()='Laufende und abgelaufene Entsendesperren"
-        + " (Art. 7 Abs. 2 Entsendegesetz)']/following::table[1]"
-    )[0]
-    for row in table.xpath("tbody/tr")[1:]:
-        [start, name, addr, law, end] = row.xpath("descendant::*/text()")
-        address = parse_address(context, addr)
+def extract_rows(path: Path) -> List[Dict[str, str]]:
+    pdf = pdfplumber.open(path.as_posix())
+    settings = {
+        "vertical_strategy": "lines",
+        "horizontal_strategy": "lines",
+        "text_tolerance": 1,
+    }
+    headers: Optional[List[str]] = None
+    rows: List[Dict[str, str]] = []
+    for page in pdf.pages:
+        cropped = page.crop((0, 50, page.width, page.height - 10))
+        for row in cropped.extract_table(settings):
```

Comment:
out of interest, was the cropping necessary, or just a bit of cleanup? extract_table returns the biggest table discovered by default, so I'd expect it to pick the table and nothing else, handling margin changes out of the box.

Reply:
It actually fixed a fixed a parsing app. The first page of the PDF contains a really heavy line at the top as a visual element, and that was used by plumber as the first cell divider line unless I cut it off.